pkg/infrastructure/services/deployer.go

```diff
@@ -55,7 +54,7 @@ type ServiceDefinition struct {
 	// E.g. if you need an additional Kubernetes resource, just create your own map that the API will append to its managed resources.
 	// The "objectLists" array is the List object reference of the types created.
 	// For example: if a ConfigMap is created, then ConfigMapList empty reference should be added to this list
-	OnObjectsCreate func(kogitoService v1alpha1.KogitoService) (resources map[reflect.Type][]resource.KubernetesResource, objectLists []runtime.Object, err error)
+	OnObjectsCreate func(cli *client.Client, kogitoService v1alpha1.KogitoService) (resources map[reflect.Type][]resource.KubernetesResource, objectLists []runtime.Object, err error)
```

Comment:
Normally this function is set by controllers, which already have a reference for the Kubernetes client. We really need to pass it as parameter?

Reply:
I passed it as argument because these functions are not define in same file as controller. For reference `onObjectsCreate` method is defined in `deployer.go` whereas controller file `kogitoruntime_controller.go`

I could use client reference from controller to call `onObjectsCreate`, in that case whether I have to define this function as 
`func (r *ReconcileKogitoRuntime) onObjectsCreate(` inside deployer.go file 
or move this function inside `kogitoruntime_controller.go`

Reply:
@vaibhavjainwiz this is a reference for a function, which is always defined on a controller, being called by the Service API. You don't need to pass a reference to it since the controllers already have a reference to it.

https://github.com/kiegroup/kogito-cloud-operator/blob/master/pkg/controller/kogitoruntime/deployer.go#L65

To make this function that I linked aware of the client reference, just add it to the `kogitoruntime` object, like:

```go
func (r *ReconcileKogitoRuntime) onObjectsCreate(kogitoService v1alpha1.KogitoService) (map[reflect.Type][]resource.KubernetesResource, []runtime.Object, error) {
   // use it here as r.client....
}
```

But I just realized that `onDeploymentsCreate` are doing the same thing. Let it be and we can refactor this later.